### DISCUSSION TITLE
Detection of slow spindles and default parm. values for spindle detection

### DISCUSSION THREAD

**Discussion by kimelit** [raphaelvallat/yasa#135] [Q&A] [answered]
Hello,
I wanted to ask about these three parameters:
'rel_pow': Relative power (= power ratio freq_sp / freq_broad).
'corr': Moving correlation between original signal and sigma-filtered signal.
'rms': Number of standard deviations above the mean of a moving root mean square of sigma-filtered signal.

My first question is whether the _corr_ and _rms_ always use sigma (11-16Hz)? Because I guess this might really be sub-optimal when I search for slow spindles (8-12Hz)

The second question is whether there are values for these parameters that could be safely used for adults' sleeping data? Without trying different settings and comparing to manual marking of spindles.

Thank you!

**Comment by raphaelvallat** [comment, marked as answer]
Hi @kimelit,

The corr and rms will be based on the frequency band defined in  the` freq_sp` parameter. of the `yasa.spindles_detect` function. The default is (12, 15) Hz but you can set lower values for slow spindles detection. Please see also this example notebook for more details on slow versus fast spindles detection: https://github.com/raphaelvallat/yasa/blob/master/notebooks/04_spindles_slow_fast.ipynb

The default thresholds should work reasonably well for adults, but please keep in mind that they have not been extensively tested in a large database. I would therefore always recommend that you do some manual comparison. 

Thanks
Raphael

**Reply by kimelit** [reply]
Thank you!

**Reply by kimelit** [reply]
Hi again Raphael,
I was wondering if perhaps you are aware of projects (that were or were not published eventually) that used the YASA spindle detection algorithm? I wanted to have a look at the parameters values (i.e., rel_pow, corr, rms) that they used.
Thanks so much!

**Reply by raphaelvallat** [reply]
Hi @kimelit! Not really unfortunately, although I do know many people use it. I just searched "YASA spindles detection" on Google Scholar and it seems like there are a few papers that may be relevant: https://scholar.google.fr/scholar?hl=en&as_sdt=0%2C5&q=yasa+spindles+detection&btnG=

**Reply by kimelit** [reply]
Thanks Raphael :) Google scholar is indeed a good friend of mine too ;)
And how did you choose the default values for the algorithm?
I have noticed that two of them are quite similar (though not exactly the same) to the original paper by Lacourse and colleagues, but there's also the rel_pow value in YASA.

**Reply by raphaelvallat** [reply]
The `rel_pow` threshold in YASA is similar to the `A7relSigPow` threshold in Lacourse et al 2018. I chose the default thresholds based on some internal tests, but this was a long time ago so I don't remember exactly how I came up with the three values. In any cases, I always recommend that users find the best combination of thresholds in their own data: https://raphaelvallat.com/yasa/build/html/faq.html#best_params

It is also possible to start with only one threshold (RMS), and then if needed, add the second and/or third threshold. The moving correlation and relative sigma power thresholds are a bit redundant so it is possible that the algorithm would work well with just two thresholds.
